rag/svr/task_executor_refactor/dataset_structure_merger.py

```
@@ -14,143 +14,468 @@
+async def _load_last_build_time(tenant_id: str, kb_id: str, compile_kwd: str, template_id: str | None) -> float | None:
+    """Read the last build timestamp from a metadata row."""
+    meta_id = _meta_row_id(kb_id, compile_kwd, template_id)
+    index = _index_name(tenant_id)
+    if not settings.docStoreConn.index_exist(index, kb_id):
+        return None
+    try:
+        row = await thread_pool_exec(settings.docStoreConn.get, meta_id, index, [kb_id])
+        if row:
+            return float(row.get("build_timestamp_flt", 0))
+    except Exception:
+        pass
```

Ruff (0.16.0)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)